Sail/src/Sail/entities/EntityFactory.cpp

```diff
@@ -56,21 +48,34 @@ Entity::SPtr EntityFactory::CreateMyPlayer(Netcode::PlayerID playerID, size_t li
 	Netcode::ComponentID netComponentID = myPlayer->getComponent<NetworkSenderComponent>()->m_id;
 	myPlayer->addComponent<NetworkReceiverComponent>(netComponentID, Netcode::EntityType::PLAYER_ENTITY);
 	myPlayer->addComponent<LocalOwnerComponent>(netComponentID);
-
-	myPlayer->addComponent<MovementComponent>()->constantAcceleration = glm::vec3(0.0f, -9.8f, 0.0f);
+	myPlayer->removeComponent<CollidableComponent>();
 	myPlayer->addComponent<CollisionComponent>();
-	myPlayer->addComponent<SpeedLimitComponent>()->maxSpeed = 6.0f;
-
-
-	// Create candle for the player
-	auto candle = CreateCandle("PlayerCandle", glm::vec3(0.f, 2.f, 0.f), lightIndex);
-	candle->addComponent<RealTimeComponent>(); // Player candle will have its position updated each frame
-	candle->getComponent<CandleComponent>()->setOwner(playerID);
-	myPlayer->addChildEntity(candle);
+	myPlayer->getComponent<ModelComponent>()->renderToGBuffer = false;
+	myPlayer->addComponent<MovementComponent>()->constantAcceleration = glm::vec3(0.0f, -9.8f, 0.0f);
 
-	myPlayer->getComponent<TransformComponent>()->setStartTranslation(glm::vec3(1.6f, 0.9f, 1.f) + spawnLocation);
+	for (Entity::SPtr& c : myPlayer->getChildEntities()) {
+		if (c->getName() == myPlayer->getName() + "Candle") {
+			c->addComponent<CandleComponent>();
```

Comment:
This code segment is shared in two places, could probably be extracted into a function